src/ApplicationInsights.Kubernetes/K8sEnvironmentFactory.cs

```diff
@@ -42,15 +43,28 @@ internal class K8sEnvironmentFactory : IK8sEnvironmentFactory
                 using (IKubeHttpClient httpClient = _httpClientFactory.Create(_httpClientSettings))
                 using (K8sQueryClient queryClient = _k8sQueryClientFactory.Create(httpClient))
                 {
-                    string containerId = _httpClientSettings.ContainerId;
+                    // TODO: See if there's better way to fetch the container id
+                    K8sPod myPod = await SpinWaitUntilGetPod(timeoutAt, queryClient).ConfigureAwait(false);
+                    string containerId = null;
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    {
+                        // For Windows, is there a way to fetch the current container id from within the container?
+                        containerId = myPod.Status.ContainerStatuses.First().ContainerID;
```

Comment:
Are you sure you will always be the first container in the pod?

Reply:
Also 
1. SpinWaitUntilGetPod() may return null
2. ContainerStatuses might be empty (low probability, but this is result of a network call, so sh*t happens)

Reply:
No, I am **not** sure. That is why I stated in the README.md that only **single container per pod** is supported for Windows container.
I **DO** want to review this later. If we could somehow get the container id from within the container itself, just like get it from cGroup in Linux, this problem will be solved automatically.

Reply:
The "COMPUTERNAME" environment variable in the container is set to a shortened version of the container id

Reply:
@david-jarman From what I see, COMPUTERNAME is set to substring of the podName than the container id, is that not the case on your cluster?

Reply:
SpinWaitUntilGetPod() may return null, meaning, on line 52 you will get NullRefException if it does